### ISSUE TITLE
Added Box2D, unable to import

### ISSUE THREAD

**Issue by jarl-haggerty** [iodide-project/pyodide#1101]
I added a packages/Box2D folder with the following meta.yaml file:

```
package:
  name: Box2D
  version: 2.3.10

source:
  url: https://github.com/pybox2d/pybox2d/archive/2.3.10.zip

```

pyodide builds successfully, `pyodide.loadPackage('Box2D')` appears to run file, but when I run `import Box2D` I get an error when Box2D tries to load a submodule:

```
pyodide.asm.js:8 Uncaught (in promise) Error: Traceback (most recent call last):
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 290, in eval_code
    return CodeRunner(ns, mode, quiet_trailing_semicolon, filename).run(code)
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 222, in run
    exec(mod, self.ns, self.ns)
  File "<exec>", line 2, in <module>
  File "/lib/python3.8/site-packages/Box2D/__init__.py", line 20, in <module>
    from .Box2D import *
ModuleNotFoundError: No module named 'Box2D.Box2D'

    at _hiwire_throw_error (pyodide.asm.js:8)
    at _pythonexc2js (:8000/pyodide.asm.wasm:wasm-function[385]:0x11352a)
    at __pyproxy_apply (:8000/pyodide.asm.wasm:wasm-function[383]:0x113227)
    at Module.__pyproxy_apply (pyodide.asm.js:8)
    at Object.apply (pyodide.asm.js:8)
    at Object.Module.runPython (pyodide.js:331)
    at (index):20
```

It sounds like some of Box2D's files are missing from the WASM file system.  Is there a step I'm missing to make sure that all the Box2D files are packaged?

**Comment by jarl-haggerty** [comment]
It looks like I have tests, examples, the top level __init__.py and the shared library installed but none of the actual library's python files.

```
pyodide.runPython("print(os.listdir('/lib/python3.8/site-packages/Box2D'))")
pyodide.asm.js:8 ['tests', 'examples', 'b2', '__init__.py', '_Box2D.so']
```

**Comment by jarl-haggerty** [comment]
I see that the `install_for_distribution(args)` function in `pyodide_build/pywasmcross.py` doesn't place `Box2D.py` in `packages/.artifacts/lib/python/Box2D-2.3.10-py3.8-linux-x86_64.egg/Box2D/`.  If I remove the skip-build argument then Box2D.py gets generated correctly.  Which is nice but I imagine this is going to replace the wasm compiled code with machine code.

**Comment by jarl-haggerty** [comment]
I posted here because I only get this error in pyodide but now I'm able to reproduce it in my system's python.  If I build from a new checkout of pybox2d AND I leave the source directly before trying to test Box2D then I get the above error.

The underlying problem appears to be that swig generates Box2D/Box2D.py after setuptools installs the .py files.  This is a problem that nlopt solves by using the cmdclass parameter in setup() to run swig before the python build.  So I've copied that solution.

https://github.com/iodide-project/pyodide/blob/3e89750d16831f5378bd2e08bb62585c92db2365/packages/nlopt/src/setup.py#L158
